schema_based_on_diagram.sql

```diff
@@ -0,0 +1,51 @@
+-- schema_based_on_diagram
+CREATE DATABASE clinic;
+
+CREATE TABLE patients(
+id INT GENERATED ALWAYS AS IDENTITY,
+name VARCHAR,
+date_of_birth DATE,
+PRIMARY KEY(id)
+);
+
+
+CREATE TABLE medical_histories(
+id INT GENERATED ALWAYS AS IDENTITY,
+admited_at TIMESTAMP,
+patient_id INT REFERENCES patients(id),
+status VARCHAR,
+PRIMARY KEY(id)
+);
+
+CREATE TABLE treatments(
+id INT GENERATED ALWAYS AS IDENTITY,
+type VARCHAR,
+name VARCHAR,
+PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices(
+id INT GENERATED ALWAYS AS IDENTITY,
+total_amount DECIMAL,
+generated_at TIMESTAMP,
+payed_at TIMESTAMP,
+medical_history_id INT REFERENCES medical_histories(id),
+PRIMARY KEY(id)
+);
+
+CREATE TABLE invoice_items(
+id INT GENERATED ALWAYS AS IDENTITY,
+unit_price DECIMAL,
+quantity INT,
+total_price DECIMAL,
+invoice_id INT REFERENCES invoices(id),
+treatment_id INT REFERENCES treatments(id),
+PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories_treatments(
+id INT GENERATED ALWAYS AS IDENTITY,
+medical_histories_id INT REFERENCES medical_histories(id),
+treatment_id INT REFERENCES treatments(id),
```

Comment:
- [ ] Please kindly create indexes for all the foreign keys in accordance to the project requirement. Indexes are a common way to enhance database performance, an index allows the database server to find and retrieve specific rows much faster. ℹ️ 

<img width="455" alt="Untitled" src="https://github.com/DianaBeki/vet_clinic/assets/103386183/8e0552ed-0c7b-4d78-9059-78d7f23a7053">